### PR TITLE
BUG make sure symlinks to nowhere do not break builds

### PR DESCRIPTION
**Pull request by beckermr**
This PR fixes an error where if I build a recipe with a symlink to a file that does not exist, then conda build chokes when it tries to get the size of the symlink.

I think this bug is breaking work on OpenMP for conda-forge here 

 conda-forge/openmp-feedstock#29

and here 

https://github.com/conda-forge/ctng-compilers-feedstock/pull/7

@isuruf @jjhelmus @marcelotrevisani  for viz

### CODE REVIEW

File: conda_build/build.py [hunk 1]
```diff
@@ -1273,9 +1273,12 @@ def build_info_files_json_v1(m, prefix, files, files_with_prefix):
         file_info = {
             "_path": short_path,
             "sha256": utils.sha256_checksum(path),
-            "size_in_bytes": os.path.getsize(path),
             "path_type": path_type(path),
         }
+        if file_info["path_type"] == PathType.hardlink:
+            file_info["size_in_bytes"] = os.stat(path).st_size
+        else:
+            file_info["size_in_bytes"] = os.lstat(path).st_size
```
Comment (marcelotrevisani) [inline conda_build/build.py]: Nice! Just one thing, do you think you can create a test just to cover this corner case? That would be nice!

**Comment by beckermr** [comment]
OK @marcelotrevisani, it turns out the file sizes of symlinks are not stable and so they are not suitable for making package hashes. Instead, I have conda-build return zero bytes and the SHA256 sum of an empty file for symlinks that point to a file that does not exist. 

I added tests of both the routines and a test where a recipe with such a symlink is built and then used to make sure it functions properly.

**Comment by beckermr** [comment]
@marcelotrevisani Shall we merge this?

**Comment by beckermr** [comment]
Bump here @marcelotrevisani. I'd be great to get this merged so we can unblock the openmp work.

File: tests/test_build.py [hunk 2]
```diff
@@ -127,6 +126,51 @@ def test_create_info_files_json(testing_workdir, testing_metadata):
         assert output == expected_output
 
 
+@pytest.mark.skipif(on_win and sys.version[:3] == "2.7",
+                    reason="os.symlink is not available so can't setup test")
+def test_create_info_files_json_symlinks(testing_workdir, testing_metadata):
```
Comment (marcelotrevisani) [inline tests/test_build.py]: Thanks for the tests! I really appreciate that! :)

**Comment by isuruf** [comment]
This is a very rare use case, but silently ignoring dangling symlinks is a bigger issue IMO. conda-build should at least give a warning about dangling symlinks.

**Comment by beckermr** [comment]
Fair point @isuruf. I made an issue here: https://github.com/conda/conda-build/issues/3843

I probably won't get to it today, but hopefully this coming week.
